src/tidal_mcp/auth.py

```
@@ -0,0 +1,558 @@
+                'response_type': 'code',
+                'client_id': self.client_id,
+                'redirect_uri': self.REDIRECT_URI,
+                'scope': 'r_usr w_usr w_sub',
+                'code_challenge': code_challenge,
+                'code_challenge_method': 'S256',
+                'state': secrets.token_urlsafe(32)
+            }
+            
+            auth_url = f"{self.OAUTH_BASE_URL}/auth?{urlencode(auth_params)}"
+            
+            print(f"\nOpening browser for Tidal authentication...")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)